cpp/CMakeLists.txt

```diff
@@ -40,299 +50,135 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
-    "$ENV{CONDA_PREFIX}/include")
-endif(NOT BUILD_STATIC_FAISS)
-message(STATUS "FAISS: FAISS_INCLUDE_DIRS set to ${FAISS_INCLUDE_DIRS}")
+if (NOT DISABLE_OPENMP OR NOT ${DISABLE_OPENMP})
+  find_package(OpenMP)
+  if(OPENMP_FOUND)
+    message(VERBOSE "RAFT: OpenMP found in ${OpenMP_CXX_INCLUDE_DIRS}")
+  endif(OPENMP_FOUND)
+endif(NOT DISABLE_OPENMP OR NOT ${DISABLE_OPENMP})
 
-###################################################################################################
-# - RMM -------------------------------------------------------------------------------------------
+# add third party dependencies using CPM
+include(cmake/thirdparty/get_cpm.cmake)
 
-find_path(RMM_INCLUDE_DIRS "rmm"
-    HINTS
-    "$ENV{RMM_ROOT}/include"
-    "$ENV{CONDA_PREFIX}/include/rmm"
-    "$ENV{CONDA_PREFIX}/include")
+# add third party dependencies using CPM
+include(cmake/thirdparty/get_cpm.cmake)
 
-message(STATUS "RMM: RMM_INCLUDE_DIRS set to ${RMM_INCLUDE_DIRS}")
+include(cmake/thirdparty/get_faiss.cmake)
 
+include(cmake/thirdparty/get_rmm.cmake)
 
-##############################################################################
-# - include paths ------------------------------------------------------------
+if(BUILD_TESTS)
+  include(cmake/thirdparty/get_nccl.cmake)
 
-set(RAFT_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include CACHE STRING
-  "Path to RAFT include directories")
+  include(cmake/thirdparty/get_ucx.cmake)
 
-set(RAFT_INCLUDE_DIRECTORIES
-  ${RAFT_INCLUDE_DIR}
-  ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}
-  ${FAISS_INCLUDE_DIRS}
-  ${RMM_INCLUDE_DIRS})
-
-if(NOT CUB_IS_PART_OF_CTK)
-  list(APPEND RAFT_INCLUDE_DIRECTORIES ${CUB_DIR}/src/cub)
-endif(NOT CUB_IS_PART_OF_CTK)
-
-if(DEFINED ENV{CONDA_PREFIX})
-  message(STATUS "Using RMM installation from $ENV{CONDA_PREFIX}")
-  list(APPEND RAFT_INCLUDE_DIRECTORIES $ENV{CONDA_PREFIX}/include)
-endif(DEFINED ENV{CONDA_PREFIX})
+  include(cmake/thirdparty/get_gtest.cmake)
+endif()
 
 ##############################################################################
-# - libraries ----------------------------------------------------------------
-
-set(RAFT_LINK_LIBRARIES
-  ${CUDA_cublas_LIBRARY}
-  ${CUDA_cusolver_LIBRARY}
-  ${CUDA_CUDART_LIBRARY}
-  ${CUDA_cusparse_LIBRARY}
-  ${CUDA_curand_LIBRARY})
-
-set(RAFT_LINK_DIRECTORIES
-  ${FAISS_INCLUDE_DIRS}
-  ${RMM_INCLUDE_DIRS})
-
-if(DEFINED ENV{CONDA_PREFIX})
-  list(APPEND RAFT_LINK_DIRECTORIES $ENV{CONDA_PREFIX}/lib)
-endif(DEFINED ENV{CONDA_PREFIX})
+# - install targets-----------------------------------------------------------
+
+add_library(raft INTERFACE)
+add_library(raft::raft ALIAS raft)
+target_include_directories(raft INTERFACE "$<BUILD_INTERFACE:${RAFT_SOURCE_DIR}/include>"
+                                          "$<INSTALL_INTERFACE:include>")
+
+target_link_libraries(raft
+INTERFACE
+  CUDA::cublas
+  CUDA::curand
+  CUDA::cusolver
+  CUDA::cudart
+  CUDA::cusparse
+  rmm::rmm
+  spdlog::spdlog_header_only
```

Comment:
`spdlog::*` and `raft::spdlog_header_only` can both be yanked. Usage of `rmm::rmm` will bring in the correct `spdlog` for you.